jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/solution/route/activity/AbstractActivityNEW.java

```diff
@@ -0,0 +1,128 @@
+package com.graphhopper.jsprit.core.problem.solution.route.activity;
+
+
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import com.graphhopper.jsprit.core.problem.Capacity;
+import com.graphhopper.jsprit.core.problem.Location;
+
+public abstract class AbstractActivityNEW implements TourActivity {
+
+    private int index;
+    protected Capacity capacity;
+    protected double arrTime;
+    protected double endTime;
+    protected double theoreticalEarliest = 0;
+    protected double theoreticalLatest = Double.MAX_VALUE;
+    protected String name;
+    protected Location location;
+
+
+    public AbstractActivityNEW(String name, Location location, Capacity capacity) {
+        super();
```

Comment:
Where do we consider that an activity can have multiple time windows, multiple locations etc.? If a job is a sequence of activities, I propose to consider this within an activity. We need to take into account that an assigned/scheduled activity can only have a single time window as it must have a single locations. However, there may be multiple time windows/locations to choose from. Maybe, we should differentiate between an ordinary activity and a scheduled activity. First holds all the information about locations, multiple time windows etc. Latter is a concrete assignment to a route and a driver.